src/azure-cli/azure/cli/command_modules/servicefabric/operations/managed_node_types.py

```diff
@@ -226,6 +232,47 @@ def add_vm_extension(cmd,
         raise
 
 
+def update_vm_extension(cmd,
+                        client,
+                        resource_group_name,
+                        cluster_name,
+                        node_type_name,
+                        extension_name,
+                        type_handler_version=None,
+                        force_update_tag=None,
+                        auto_upgrade_minor_version=None,
+                        setting=None,
+                        protected_setting=None,
+                        provision_after_extension=None,
+                        setup_order=None):
+    try:
+        node_type: NodeType = client.node_types.get(resource_group_name, cluster_name, node_type_name)
+        existing_extension = find_in_collection(node_type, 'vm_extensions', 'name', extension_name)
+
+        if existing_extension is None:
+            logger.error('Extension %s does not exist.', extension_name)
+            return None
+
+        newExtension = VMSSExtension(name=extension_name,
+                                     publisher=existing_extension.publisher,
+                                     type=existing_extension.type,
+                                     type_handler_version=type_handler_version if type_handler_version is not None else existing_extension.type_handler_version,
+                                     force_update_tag=force_update_tag if force_update_tag is not None else existing_extension.force_update_tag,
+                                     auto_upgrade_minor_version=auto_upgrade_minor_version if auto_upgrade_minor_version is not None else existing_extension.auto_upgrade_minor_version,
+                                     settings=setting if setting is not None else existing_extension.settings,
+                                     protected_settings=protected_setting if protected_setting is not None else existing_extension.protected_settings,
```

Comment:
Protected settings are not returned on the get but I am not sure what is the value here so If they are not passed in on the update, I would say keep it as None so they don't get reset with empty value or something like that. And can you validate they don't change in that case? you try it using a custom script extension.

Reply:
updated it